wrfhydropy/core/job.py

```diff
@@ -307,8 +333,27 @@ def _set_hrldas_times(self):
 
                 self._hrldas_times['noahlsm_offline']['restart_filename_requested'] = lsm_restart_file
 
-            self._hrldas_times['noahlsm_offline']['restart_frequency_hours'] = self.restart_freq_hr
-            self._hrldas_times['noahlsm_offline']['output_timestep'] = self.output_freq_hr * 3600
+            # TODO(JLM): I dont love this if statement, it's a bit hacky.
+            # Some of the tests call _set_hrldas/hydro_times when no namelist has been set.
+            # That's why this is here.
+            if self._hrldas_namelist is not None:
+                noah_nlst = self._hrldas_namelist['noahlsm_offline']
+
+                the_noahlsm_offline = self._hrldas_times['noahlsm_offline']
+                if self.restart_freq_hr_hrldas is not None:
+                    if self.restart_freq_hr_hrldas > 0:
+                        the_noahlsm_offline['restart_frequency_hours'] = self.restart_freq_hr_hrldas
+                    else:
+                        the_noahlsm_offline['restart_frequency_hours'] = -99999
+                else:
+                    the_noahlsm_offline['restart_frequency_hours'] = \
+                        noah_nlst['restart_frequency_hours']
+
+                if self.output_freq_hr_hrldas is not None:
+                    the_noahlsm_offline['output_timestep'] = self.output_freq_hr_hrldas * 3600
+                else:
+                    the_noahlsm_offline['output_timestep'] = noah_nlst['output_timestep']
+
```

Comment:
Our default behavior is getting kind of out of hand, its just this I've put it everywhere also. I am wondering if we should revisit how we do this and make it more rigid. It would require more setup by the user but would be more clear, which is a tradeoff I am generally in favor of

Reply:
We are going to document the behavior in the docstring.